src/persist-client/src/impl/machine.rs

```diff
@@ -656,6 +661,10 @@ where
         debug_assert!(self.state.seqno() <= current.seqno);
         self.state = current_state;
     }
+
+    pub fn seqno(&self) -> SeqNo {
```

Comment:
nit: I'd put this next to `fn upper()` and similar methods. Just to group these simple accessors together.

Reply:
Done in latest push